src/states_screens/dialogs/ghost_replay_info_dialog.cpp

```diff
@@ -59,17 +96,98 @@ GhostReplayInfoDialog::GhostReplayInfoDialog(unsigned int replay_id)
     }
 
     m_record_widget->setState(false);
-    m_watch_widget->setState(false);
+    m_watch_widget->setState(m_compare_ghost);
+    m_compare_widget->setState(m_compare_ghost);
+
+    if (m_compare_ghost)
+    {
+        m_watch_only = true;
+        m_record_race = false;
+        m_record_widget->setState(false);
+        m_record_widget->setVisible(!m_watch_only);
+        getWidget<LabelWidget>("record-race-text")->setVisible(!m_watch_only);
+    }
+
+    // Display this checkbox only if there is another replay file to compare with
+    getWidget<LabelWidget>("compare-ghost-text")->setVisible(m_compare_ghost);
+    m_compare_widget->setVisible(m_compare_ghost);
+
 
     m_action_widget->setFocusForPlayer(PLAYER_ID_GAME_MASTER);
     m_action_widget->select("start", PLAYER_ID_GAME_MASTER);
 }   // GhostReplayInfoDialog
-
 // -----------------------------------------------------------------------------
 GhostReplayInfoDialog::~GhostReplayInfoDialog()
 {
 }   // ~GhostReplayInfoDialog
 
+// -----------------------------------------------------------------------------
+void GhostReplayInfoDialog::updateReplayDisplayedInfo()
+{
+    m_replay_info_widget->clear();
+
+    int n=2;
```

Comment:
Can you maybe find a more descriptive name that "n" so it's easier to understand what it is?